src/Time/StepChoosers/StepChooser.hpp

```diff
@@ -18,84 +19,76 @@ namespace Parallel {
 template <typename Metavariables>
 class GlobalCache;
 }  // namespace Parallel
-template <typename StepChooserRegistrars>
-class StepChooser;
 // IWYU pragma: no_forward_declare db::DataBox
-template <typename StepChooserRegistrars>
-class StepChooser;
 /// \endcond
 
+/// The intended use for a step chooser.  This is used to control the
+/// classes via factories.
+namespace StepChooserUse {
+struct Slab;
+struct LtsStep;
```

Comment:
Can these be enum cases? If no, could you add a "design decision" comment why they are not? Perhaps you could use keep the existing `UsableFor` enum

The cases/structs should probably also have a short docstring.

Reply:
I think I did it this way to make forward declarations easier, but nothing forward declares any of this stuff so that seems silly in hindsight.

Reply:
I'm not certain it's relevent in any current metafunctions, but templates on `typename` tend to be better behaved for use in brigand metafunctions (because of `typename...` packs).
Now that we're using more brigand metafunctions to track these class lists instead of the registration, it seems potentially more convenient for future utilities to have mostly types as the template parameters.

Reply:
Ok. It's a bit inconvenient to have to resort to this sort of thing - I am often annoyed by some of the DataBox prefixes like Tags::deriv taking the Dim as a tmpl::size_t instead of a size_t. Up to you want you think is best in this case. However, would you mind adding a brief "design decision" comment to the code?